libs/@hashintel/petrinaut/src/views/Editor/components/PropertiesPanel/transition-properties.tsx

```diff
@@ -56,6 +57,13 @@ export const TransitionProperties: React.FC<TransitionPropertiesProps> = ({
   updateTransition,
   onArcWeightUpdate,
 }) => {
+  const simulationState = useSimulationStore((state) => state.state);
+
+  // Check if simulation is running or paused
+  const isSimulationActive =
+    simulationState === "Running" || simulationState === "Paused";
+  const isReadOnly = globalMode === "simulate" || isSimulationActive;
```

Comment:
We repeat this logic in many places. Can we extract it into one of the stores so we only have to update it in one place if the readonly logic changes again?

Reply:
I put that in SimulationStore for now, because another big PR would create conflicts in one way or another.

Reply:
I did it through a hook in the end.  
Zustand does not seem designed for defining derived data in the stores.